src/libopenrave/kinbodyjoint.cpp

```diff
@@ -2035,7 +2035,10 @@ void KinBody::Joint::SetMimicEquations(int iaxis, const std::string& poseq, cons
         size_t index = eq.find('|');
         std::string sCommonSubexpressions = (index != std::string::npos) ? eq.substr(0, index) : ""; ///< common subexpressions
         if( !sCommonSubexpressions.empty() ) {
-            sCommonSubexpressions += ';'; // just in case, separate out the equations
+            const size_t semicolonindex = sCommonSubexpressions.find_last_of(";"); // avoid double semicolons
```

Comment:
@rdiankov  Meat is here.

Reply:
@tgn3000 i was aware we were potentially doing this, but i thought fparse would handle this case and basically not evaluate anything. what broke?

Reply:
@tgn3000 ping

Reply:
@rdiankov What's up, Rosen. Please see below my comments. Many thanks!